### PR TITLE
Add geyser plugin runner

### PR DESCRIPTION
**Pull request by gagliardetto**
This PR adds the rust code that can be used to read old-faithful CAR files and send their contents to a Solana geyser plugin.

This PR provides the rust libraries to read CAR files, and a CLI to dum a CAR file to a geyser plugin.

### CODE REVIEW

**Review by fanatid — COMMENTED** [review]
Can we use workspaces here? https://doc.rust-lang.org/cargo/reference/workspaces.html
would be good to have `rust-toolchain.toml` so rust version would be fixed https://rust-lang.github.io/rustup/overrides.html

File: geyser-plugin-runner/demo-plugin/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+
+const BANNER: &str = "::plugin::";
+
+impl GeyserPlugin for GeyserPluginDemo {
+    fn name(&self) -> &'static str {
+        "plugin::GeyserPluginDemo"
+    }
+
+    fn on_load(&mut self, config_file: &str) -> Result<()> {
+        println!(
```
[the remark this answers is not in the document]
Comment (gagliardetto) [reply geyser-plugin-runner/demo-plugin/src/lib.rs]: fixed

File: geyser-plugin-runner/demo-plugin/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+impl std::fmt::Debug for GeyserPluginDemo {
+    fn fmt(&self, _: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        Ok(())
+    }
+}
+
+fn green_bg(s: &str) -> String {
+    // green BG, black FG
+    format!("\x1b[42;30m{}\x1b[0m", s)
+}
```
[the remark this answers is not in the document]
Comment (gagliardetto) [reply geyser-plugin-runner/demo-plugin/src/lib.rs]: Used.

File: geyser-plugin-runner/src/dataframe.rs [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+            hash: None,
+            index: None,
+            total: None,
+            data: Buffer::new(),
+            next: None,
+        };
+
+        if let serde_cbor::Value::Array(array) = val {
+            // println!("Kind: {:?}", array[0]);
+            if let serde_cbor::Value::Integer(kind) = array[0] {
```
[the remark this answers is not in the document]
Comment (gagliardetto) [reply geyser-plugin-runner/src/dataframe.rs]: Fixed everywhere.

**Review by linuskendall — APPROVED** [review]

**Pull request merged**